src/ui/starter-select-ui-handler.ts

```diff
@@ -1873,12 +1882,40 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
                 this.pokemonShinyIcon.setTint(tint);
                 this.pokemonShinyIcon.setVisible(true);
               } else {
-                this.setSpeciesDetails(this.lastSpecies, false, undefined, undefined, 0, undefined, undefined);
+                this.setSpeciesDetails(this.lastSpecies, { shiny: false, variant: 0 });
                 this.pokemonShinyIcon.setVisible(false);
                 success = true;
               }
             }
             break;
+          case Button.V:
+            if (this.canCycleVariant) {
+              let newVariant = props.variant;
+              do {
+                newVariant = (newVariant + 1) % 3;
+                if (!newVariant) {
```

Comment:
```suggestion
                if (newVariant === 0) {
```
for clarity. This could really be a switch case but that would likely end up confusing with the breaks